packages/eventindexer/nft_balance.go

```diff
@@ -30,8 +30,11 @@ type UpdateNFTBalanceOpts struct {
 
 // NFTBalanceRepository is used to interact with nft balances in the store
 type NFTBalanceRepository interface {
-	SubtractBalance(ctx context.Context, opts UpdateNFTBalanceOpts) (*NFTBalance, error)
-	IncreaseBalance(ctx context.Context, opts UpdateNFTBalanceOpts) (*NFTBalance, error)
+	IncreaseAndSubtractBalancesInTx(
+		ctx context.Context,
+		increaseOpts UpdateNFTBalanceOpts,
+		subtractOpts UpdateNFTBalanceOpts,
+	) (*NFTBalance, *NFTBalance, error)
```

Comment:
returning two pointers to the same strict here is not clear. we should either give them names return parameters, or wrap them in a parent struct and give each a named member, like "IncreasedNFTBalance" and "DecreasedNFTBalance" on the parent struct of "NFTBalances"